### ISSUE TITLE
Add BenchmarkDotNet triggers on MonoJenkins bot

### ISSUE THREAD

**Issue by alexanderkyte** [mono/mono#11689]
It would be useful to have some triggers to invoke the microbenchmark jobs. 

`@monojenkins benchmark <arch>` would kick off a benchmark on the `arch` configuration.

Since BenchmarkDotNet makes it possible to run a benchmark from a gist, I think we should try to expose that transparently. That will make it easy for people to invoke benchmarks without having to interact with and understand the underlying benchmarking infrastructure. 

The command would look like

`@monojenkins benchmark <url> <arch>`

where url is a gist url, and arch is an optional name of a lane to run the benchmark on. 

For a default arch, what makes the most sense? In our upcoming work, this will probably be fairly AOT centric. 

Defaulting to linux+x64+JIT is probably still alright though. This is a general-purpose benchmarking suite integration. This isn't just for working on the JIT. I expect this to be useful for managed work as well.

**Comment by alexanderkyte** [comment]
We can certainly make it look uniform with the command to invoke a test run. 

@luhenry would you envision it looking like 

`@monojenkins build <arch> Benchmark <url>` then?

**Comment by alexanderkyte** [comment]
It sounds like you're planning to have a new configuration for the benchmark, I was going to try to just reuse what's already there and change what steps the jenkins script runs when we're benchmarking. I'm sure there are advantages to making new lanes. 

This seems like design work that @akoeplinger probably has stronger opinions on, since it has a larger impact more on how the jenkins side of things are set up.

**Comment by luhenry** [comment]
@akoeplinger could you please take a look at it?

**Comment by akoeplinger** [comment]
So there are two requests in here:

1) Trigger benchmarks for different arches
2) Trigger a benchmark which runs a benchmark from a gist url

For 2) this should be relatively easy since the pull request builder plugin exposes the comment that triggered the job in the `ghprbCommentBody` environment variable and you could extract the URL from there.

For 1) we need a separate PR job for each architecture/config we want to support so to keep it reasonable I'd propose we choose a few interesting combos.

**Comment by alexanderkyte** [comment]
The linked PR being merged should unblock this issue: https://github.com/mono/mono/pull/9833

**Comment by akoeplinger** [comment]
No longer relevant with .NET5 as we have the infrastructure there.
